.github/workflows/CI_build.yml

```diff
@@ -87,6 +87,67 @@ jobs:
           Write-Output "Run standard jobs"
         }
 
+        if ($commit_title -match "\[force nopythoncache\]") {
+          $env:PYTHON_ALLOW_CACHE = "false"
+          Write-Output "PYTHON_ALLOW_CACHE=false" >> $env:GITHUB_ENV
+        }
+        if ($Env:PYTHON_ALLOW_CACHE -eq "true") {
+          $python = ((python -V) -split " ")[1]
+          $requests = (Invoke-RestMethod https://pypi.org/pypi/requests/json).info.version
+          $rfc3987 = (Invoke-RestMethod https://pypi.org/pypi/rfc3987/json).info.version
+          $pywin32 = (Invoke-RestMethod https://pypi.org/pypi/pywin32/json).info.version
+          $lxml = (Invoke-RestMethod https://pypi.org/pypi/lxml/json).info.version
+          $key = "${{ runner.os }}-python_$python-requests_$requests-rfc3987_$rfc3987-pywin32_$pywin32-lxml_$lxml"
+          Write-Output "python=$key" >> $env:GITHUB_OUTPUT
+        }
+
+    - name: (cache) Lookup Python modules
```

Comment:
You know the [actions/setup-python] Action has caching built in, right?

All you would need is a few lines:

~~~yml
  - name: Set up python
    uses: actions/setup-python@v5
    with:
      python-version: '3.12'
      cache: 'pip'
~~~

This way you're not stuck without whatever default Python version the Windows runner comes with, in case it falls [out of date].


[actions/setup-python]: https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies
[out of date]: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14534#issuecomment-1872514104

Reply:
But it allows us to specify for which branches/events it should work for (I'm mainly concerned about saving)? I know that there are ready-made actions, but they usually have limited configuration options. For this reason I do not use the cache action itself, but separate save and restore.